File: app/src/main/java/data/vocab/models/SearchProviders.java
package data.vocab.models;

import data.vocab.jp.search.sanseido.Sanseido;

/**
 * List all available search providers and a method to return
 * the class which defines all specific implementations of models
 * related to that search provider.
 */
public class SearchProviders {
    private static final Class[] PROVIDERS = {Sanseido.class};

    /**
     * Returns the Class given the the class's name.
     * @param className the name of the provider class to return.
     * @return the Class associated with the given name if it exists, else null
     */
    public static Class getClassByKey(String className){
        for(Class provider : PROVIDERS){
            if(provider.getName().equals(className)){
                return provider;
            }
        }
        return null;
    }
}

File: app/src/main/java/data/vocab/shared/SearchProviders.kt
package data.vocab.shared

import data.vocab.lang.jp.sanseidou.Sanseidou

/**
 * List all available search providers and a method to return
 * the class which defines all specific implementations of models
 * related to that search provider.
 */
object SearchProviders {
    private val PROVIDERS = arrayOf<Class<*>>(Sanseidou::class.java)

    /**
     * Returns the Class given the the class's name.
     * @param className the name of the provider class to return.
     * @return the Class associated with the given name if it exists, else null
     */
    fun getClassByKey(className: String): Class<*>? {
        for (provider in PROVIDERS) {
            if (provider.name == className) {
                return provider
            }
        }
        return null
    }
}
